src/main/java/com/coscale/sdk/client/ApiClient.java

```diff
@@ -328,6 +328,7 @@ private void login() throws IOException {
             } catch (CoscaleApiException e) {
                 if (e.statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
                     this.token = null; // will trigger new login
+                    responseCode = 401; // 401 means login failed: https://httpstatuses.com/401
```

Comment:
It used to be that the responseCode remained 0, which requires users to read the actual error message before knowing what's wrong.